docs/src/userguide/fusedtypes.rst

```diff
@@ -162,24 +289,56 @@ by calling.
 Indexing
 --------
 
-You can index functions with types to get certain specializations, i.e.::
+You can index functions with types to get certain specializations, i.e.:
```

Comment:
After longer investigation I found out that *only* `def` and `cpdef` functions can be called via indexing. Regular `def` classes cannot be called from cython. Following example fails to compile:
```cython
cimport cython

ctypedef fused fused_type1:
    double
    float
ctypedef fused fused_type2:
    double
    float
def func(fused_type1 a, fused_type2 b):
    print("func called:", cython.typeof(a), a, cython.typeof(b), b)

# both index calls to function fails during compilation.
func[float, double](1.0, 2.0)
func[cython.float, cython.double](1.0, 2.0)
```

Reply:
Indexing with strings does work:

```
func["float", "double"](1.0, 2.0)
```

I don't think it's a bug but it maybe documentation could be improved. Essentially indexing a `def` function is purely a run-time operation - Cython doesn't try to understand it. Therefore, it need to be indexed with actual Python objects. Where these fail is that `double`, `cython.float`, and `cython.double` don't have a Python object associated with them.

(Haven't read the fused types userguide recently, or the rest of this PR, so not commenting on where the documentation would fit)

Reply:
Ok let me put this to user guide because it is not documented. It is surprise for that it can be done.

Reply:
Fixed in documentation